src/hooks/useFlag.js

```diff
@@ -1,8 +1,20 @@
-// TODO: Write useFlag hook
+import compareAnswers from "../utils/compareAnswers";
+import { setFlagTrue } from "../redux/missions";
+import { useSelector, useDispatch } from "react-redux";
+import { addRecord } from "./useTerminal";
 
-// checkFlag: (state, { value, missionId, flagId }) => {
-//   // 1. Get mission and flag
-//   // 2. Hash value with md5
-//   // 3. Check if value is correct with saved falg
-//   // 4. If correct, set "done" to true
-// },
+const useFlag = (missionIndex, flagIndex, userAnswer) => {
```

Comment:
If we use the global state of missions (missionsStore), there is no need to pass those args. 
Especially `userAnswer` - that's rather a parameter for the checkFlag function with should be returned by the hook. 

Btw, you can imagine hooks as a initializers, which can return values and functions to use in component, like:
```js
...
const [getBaked, highRate] = useWeed();
return (
  <div>
    <button onClick={() => getBaked(amount)}>smoke</button>
    <p> {highRate} / 10 </p>
  </div>
)
...
```

Hope it helps